tests/compatibility/start.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+set -eu
+
+cur=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+source $cur/../_utils/test_prepare
+WORK_DIR=$TEST_DIR/$TEST_NAME
+
+function run() {
+    run_sql_file $cur/data/db1.prepare.sql $MYSQL_HOST1 $MYSQL_PORT1
+    check_contains 'Query OK, 2 rows affected'
+    run_sql_file $cur/data/db2.prepare.sql $MYSQL_HOST2 $MYSQL_PORT2
+    check_contains 'Query OK, 3 rows affected'
+
+    echo "use previous dm-master and dm-worker"
+    run_dm_worker $WORK_DIR/worker1 $WORKER1_PORT $cur/conf/dm-worker1.toml previous
+    check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER1_PORT 
+    run_dm_worker $WORK_DIR/worker2 $WORKER2_PORT $cur/conf/dm-worker2.toml previous
+    check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER2_PORT
+    run_dm_master $WORK_DIR/master $MASTER_PORT $cur/conf/dm-master.toml previous
+    check_rpc_alive $cur/../bin/check_master_online 127.0.0.1:$MASTER_PORT
+
+    echo "start DM task only"
+    dmctl_start_task
+
+    echo "use sync_diff_inspector to check full dump loader"
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+
+    pkill -hup dm-worker.test.previous 2>/dev/null || true
+    wait_process_exit dm-worker.test.previous
+
+    echo "restart dm-worker, one use current version, and another one use previous version"
+    run_dm_worker $WORK_DIR/worker1 $WORKER1_PORT $cur/conf/dm-worker1.toml current
+    run_dm_worker $WORK_DIR/worker2 $WORKER2_PORT $cur/conf/dm-worker2.toml previous
+
+    run_sql_file $cur/data/db1.increment.1.sql $MYSQL_HOST1 $MYSQL_PORT1
+    run_sql_file $cur/data/db2.increment.1.sql $MYSQL_HOST2 $MYSQL_PORT2
+
+    echo "use sync_diff_inspector to check increment data first time "
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+    
+    echo "use current dm-master"
+    pkill -hup dm-master.test.previous 2>/dev/null || true
+    wait_process_exit dm-master.test.previous
+    run_dm_master $WORK_DIR/master $MASTER_PORT $cur/conf/dm-master.toml current
+    check_rpc_alive $cur/../bin/check_master_online 127.0.0.1:$MASTER_PORT
+    sleep 5
```

Comment:
Is this necessary or a little value is fine?

Reply:
decrease to 3